src/building/bootstrapping.md

```diff
@@ -122,50 +124,109 @@ contribution [here][bootstrap-build].
 
 ## Understanding stages of bootstrap
 
-This is a detailed look into the separate bootstrap stages. When running
-`x.py` you will see output such as:
-
-```txt
-Building stage0 std artifacts
-Copying stage0 std from stage0
-Building stage0 compiler artifacts
-Copying stage0 rustc from stage0
-Building LLVM for x86_64-apple-darwin
-Building stage0 codegen artifacts
-Assembling stage1 compiler
-Building stage1 std artifacts
-Copying stage1 std from stage1
-Building stage1 compiler artifacts
-Copying stage1 rustc from stage1
-Building stage1 codegen artifacts
-Assembling stage2 compiler
-Uplifting stage1 std
-Copying stage2 std from stage1
-Generating unstable book md files
-Building stage0 tool unstable-book-gen
-Building stage0 tool rustbook
-Documenting standalone
-Building rustdoc for stage2
-Documenting book redirect pages
-Documenting stage2 std
-Building rustdoc for stage1
-Documenting stage2 whitelisted compiler
-Documenting stage2 compiler
-Documenting stage2 rustdoc
-Documenting error index
-Uplifting stage1 rustc
-Copying stage2 rustc from stage1
-Building stage2 tool error_index_generator
-```
+### Overview
+
+This is a detailed look into the separate bootstrap stages.
+
+The convention `x.py` uses is that:
+- A `--stage N` flag means to run the stage N compiler (`stageN/rustc`).
+- A "stage N artifact" is an artifact that is _produced_ by the stage N compiler.
```

Comment:
Maybe this?

```suggestion
- A "stage N artifact" is a build that is _produced_ by the stage N compiler.
```

Reply:
How about 'a build artifact'? 'build' seems ambiguous to me, although I agree 'artifact' is not very clear on its own.

Reply:
Hmm, I think "build" is clearer by itself. What does "build artifact" add?

Another option: "is a binary"

Reply:
It is _not_ just a binary: artifacts include shared object files and .rlibs, like the standard library. I'll add that.

Reply:
I think 'build' is ambiguous because it's normally used as a verb, not a noun. Saying you've built a build seems confusing.

Reply:
I added a section on build artifacts - does it help at all?